### PR TITLE
add the processing part of the life-of-an-entity

### PR DESCRIPTION
**Pull request by freben**

### CODE REVIEW

File: docs/features/software-catalog/life-of-an-entity.md [hunk 1]
```diff
@@ -111,18 +111,55 @@ as possible.
...
 processed entity, and errors and relations to other entities separately. Then,
 the catalog checks to see what entities are touched by that output, and triggers
 the final assembly of those (see Stitching below).
 
-There are several stages involved in the processing.
-
-> TODO: More info here
+![Processing overview](../../assets/features/catalog/life-of-an-entity_processing.svg)
+
+Entities are always processed one by one. Note how each processor can contribute
```
Comment (benjdlambert) [inline docs/features/software-catalog/life-of-an-entity.md:122]: I wonder if there's some better wording here, this makes it sound like it's not distributed and we don't parallelize. But I'm struggling to think of a better way to word it.

Comment (freben) [reply docs/features/software-catalog/life-of-an-entity.md]: aight that makes sense, tweaked wording

**Comment by freben** [comment]
OK did the stitching step too

File: docs/features/software-catalog/life-of-an-entity.md [hunk 2]
```diff
@@ -111,20 +111,74 @@ as possible.
...
+
+When the final step has completed, and no errors were encountered, the processed
+entity and all of the relations are finally persisted in the database. Then the
+catalog considers this entity, and all of the entities it had relations to,
+subject for stitching.
+
+It is worth noting here that the processing does not lead to deletion or
+unregistration of entities; it can only call new entities into existence or
+update entities that it has previously called into existence. More about that
+later.
```
Comment (timbonicus) [inline docs/features/software-catalog/life-of-an-entity.md:156]: > More about that later.

This promise seems unfulfilled 😄  Remove this comment? Or add a TODO?

**Review by timbonicus — APPROVED** [review]

File: docs/features/software-catalog/life-of-an-entity.md (hunk 2 above)
Comment (freben) [reply docs/features/software-catalog/life-of-an-entity.md:156]: Maybe I meant later in time, not in space?! :) aight I'll add the todo before merging

**Pull request merged**